packages/wdio-allure-reporter/tests/cucumber.suite.test.ts

```diff
@@ -421,39 +439,37 @@ describe('reporter option "useCucumberStepReporter" set to true', () => {
             reporter.onHookEnd(cucumberHelper.hookFail())
             reporter.onTestStart(cucumberHelper.testStart())
             reporter.onTestSkip(cucumberHelper.testSkipped())
+
             const suiteResults: any = { tests: [cucumberHelper.testSkipped()], hooks: [cucumberHelper.hookFail()] }
+
             reporter.onSuiteEnd(cucumberHelper.scenarioEnd(suiteResults))
             reporter.onSuiteEnd(cucumberHelper.featureEnd(suiteResults))
             reporter.onRunnerEnd(runnerEnd())
-            const results = getResults(outputDir)
-            expect(results).toHaveLength(1)
-            allureXml = results[0]
 
-            expect(allureXml('ns2\\:test-suite > name').text()).toEqual('MyFeature')
-            expect(allureXml('test-case > name').text()).toEqual('MyScenario')
-            expect(allureXml('test-case').attr('status')).toEqual('failed')
-            expect(allureXml('step > name').eq(0).text()).toEqual('Hook')
-            expect(allureXml('step').eq(0).attr('status')).toEqual('failed')
-            expect(allureXml('step').eq(1).attr('status')).toEqual('canceled')
-            expect(allureXml('test-case parameter[kind="argument"]')).toHaveLength(1)
-            expect(allureXml('test-case parameter[name="browser"]').eq(0).attr('value')).toEqual('chrome-68')
-        })
+            const { results, containers } = getResults(outputDir)
 
-        it('should not call endStep if currentStep is not `Step` instance', () => {
-            reporter._allure.getCurrentSuite = vi.fn()
-            reporter._allure.endStep = vi.fn()
-            reporter._allure.endCase = vi.fn()
-            reporter._allure.addAttachment = vi.fn()
+            allureResult = results[0]
+            allureContainer = containers[0]
 
-            reporter.onTestFail(cucumberHelper.testSkipped())
-            expect(reporter._allure.endStep).not.toHaveBeenCalled()
-            expect(reporter._allure.endCase).toHaveBeenCalled()
+            const browserParameter = allureResult.parameters.find((param: Parameter) => param.name === 'browser')
+
+            expect(results).toHaveLength(1)
```

Comment:
please assert length before you get value

Reply:
:+1: